### PR TITLE
Editor tabs

### PR DESCRIPTION
**Pull request by 2xsaiko**
This PR mostly adds editor tabs so you can have multiple files open in parallel.

I ended up rewriting some other parts that I would have had to rewrite in some capacity anyways because of the tabbed editor, such as the top panel, and added a system for user input validation that I made most user inputs use now as opposed to throwing exceptions such as `InvalidNameException`.

This also includes a more abstracted way of decompiling classes that I wrote because the existing system was designed for having only one class open at a time.

To do:
- [x] Add close button for tabs
- [x] Fix freeze that happens when opening a class for the first time - related to `setContentType("text/enigma-sources")`
- [x] Move ClassHandle and related types out of gui package
- [x] Fix tab title bar not updating after renaming class
- [x] Renames across files don't refresh the changed name in the other file
- [x] Re-add line number bar in the editor that went... somewhere?
- [x] Add Close All/Close Others/... tab actions
- [x] Opening a class causes the cursor to appear at the bottom
- [x] Fix crash when loading `Entity`

### CODE REVIEW

File: src/main/java/cuchaz/enigma/gui/EnigmaSyntaxKit.java [hunk 1]
```diff
@@ -1,44 +1,52 @@
...
+		// This is an action written in javascript that is useless for enigma's
+		// use case, and removing it causes the editor to load way faster the
+		// first time
+		configuration.remove("Action.insert-date");
+	}
+
+	public static void invalidate() {
+		configuration = null;
+	}
+
```
Comment (Runemoro) [inline src/main/java/cuchaz/enigma/gui/EnigmaSyntaxKit.java:51]: What happened here?

File: src/main/java/cuchaz/enigma/gui/elements/CovertTextField.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+import cuchaz.enigma.gui.events.CovertTextFieldListener;
+import cuchaz.enigma.utils.Utils;
+import cuchaz.enigma.utils.validation.ParameterizedMessage;
+import cuchaz.enigma.utils.validation.Validatable;
+
+/**
+ * A label that converts into an editable text field when you click it.
+ */
+public class CovertTextField implements Validatable {
```
Comment (Runemoro) [inline src/main/java/cuchaz/enigma/gui/elements/CovertTextField.java:21]: There's a typo here. Also, I think `Converting` would be better than `Convert`.

Comment (2xsaiko) [reply src/main/java/cuchaz/enigma/gui/elements/CovertTextField.java:21]: No, that's not a typo, it's just a horrible name for what this has become (well, it has always been a horrible name). `Converting` is good actually, I think I'll go with that then.

File: src/main/java/cuchaz/enigma/gui/EnigmaSyntaxKit.java (hunk 1 above)
Comment (2xsaiko) [reply src/main/java/cuchaz/enigma/gui/EnigmaSyntaxKit.java:51]: Reformatted to use tabs like the rest of the code, I think.

Comment (Runemoro) [reply src/main/java/cuchaz/enigma/gui/EnigmaSyntaxKit.java:51]: I think most of the new classes that have been added use spaces, though. We should probably decide on one format and reformat everything.

Comment (2xsaiko) [reply src/main/java/cuchaz/enigma/gui/EnigmaSyntaxKit.java:51]: Alright, then I'll revert the formatting to spaces for this class. I'll probably not modify this class anymore anyway (IDEA replaces spaces with tabs when formatting)

Comment (modmuss50) [reply src/main/java/cuchaz/enigma/gui/EnigmaSyntaxKit.java:51]: I would suggest using checkstyle with almost identical rules to fabric and loom.

**Pull request merged**